utils_linux.go

```diff
@@ -171,6 +171,9 @@ func createPidFile(path string, process *libcontainer.Process) error {
 	if err != nil {
 		return err
 	}
+	if _, err := os.Stat(path); err == nil {
+		return fmt.Errorf("file %s already exists", path)
```

Comment:
with respect to d_type, I'm not sure whether d_type supports in all file systems, if the file system is xfs ?